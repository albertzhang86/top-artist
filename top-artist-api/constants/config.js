module.exports = {
  APIKEY: '2445f353a01f3e69b7896f73e59141b2'
};
