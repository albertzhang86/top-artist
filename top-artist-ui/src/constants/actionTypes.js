export const SEARCH_COUNTRY_SET = 'SEARCH_COUNTRY_SET';
export const SEARCH_MBID_SET = 'SEARCH_MBID_SET'
export const SEARCHED_TOP_ARTISTS = 'SEARCHED_TOP_ARTISTS';
export const RECEIVED_TOP_ARTISTS = 'RECEIVED_TOP_ARTISTS';
export const REQUESTED_TOP_ALBUMS = 'REQUESTED_TOP_ALBUMS';
export const RECEIVED_TOP_ALBUMS = 'RECEIVED_TOP_ALBUMS';
export const CLEARED_SEARCH_RESULTS = 'CLEARED_SEARCH_RESULTS';
