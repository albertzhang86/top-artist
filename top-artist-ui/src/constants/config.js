export const ITEMS_PER_PAGE = 5;
export const SEARCH_DEBOUNCE = 800;
