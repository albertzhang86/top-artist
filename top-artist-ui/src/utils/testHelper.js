import React from 'react';
import { merge } from 'lodash';

/* eslint-disable import/prefer-default-export */
export const renderComponentFactory = (Component, defaultProps, options) => {
    const renderComponent = (renderFunction, customProps = {}) => {
        const props = merge({}, defaultProps, customProps);

        return renderFunction(<Component {...props} />, options);
    };

    return renderComponent;
};
