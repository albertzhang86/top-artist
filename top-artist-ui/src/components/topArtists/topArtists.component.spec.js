import React from 'react';
import { mount, render, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import TopArtistsComponent from './topArtists.component';
import { renderComponentFactory } from '../../utils/testHelper';
import * as actionTypes from '../../constants/actionTypes';

const mockSearchTopArtists = jest.fn();

const renderComponent = renderComponentFactory(TopArtistsComponent, {
    searchTopArtists: mockSearchTopArtists,
    query: {
        page: 1,
        country: 'china'
    },
    topArtists: {
      artists: [{
        name: 'ar1',
        listeners: 2,
        mbid: '1',
        image: [{}, {}, {}, {"#test": 'http://mock.png'}]
      }, {
        name: 'ar2',
        listeners: 3,
        mbid: '2',
        image: [{}, {}, {}, {"#test": 'http://mock1.png'}]
      }],
      attr: {
        page: 1,
        totalPages: 3
      }
    },
});

describe('TopArtists', () => {
    it('renders correctly', () => {
        const component = renderComponent(renderer.create);

        expect(component).toMatchSnapshot();

        component.unmount();
    });

    it('query top artist when mount', () => {
        const component = renderComponent(shallow);
        expect(mockSearchTopArtists).toHaveBeenCalledWith({
            page: 1,
            country: 'china'
        });
        component.unmount();
    });

    it('query top artist when query changed', () => {
        const component = renderComponent(shallow);

        component.setProps({query: {
            page: 2,
            country: 'china'
        }})

        expect(mockSearchTopArtists).toHaveBeenCalledWith(
          {
              page: 2,
              country: 'china'
          }
        );
        component.unmount();
    });
    it('query top artist when search input changed', () => {
       const component = renderComponent(shallow);

       component.find('.form-control').simulate('change', {
           target: {
             value: 'spain'
           }
       });
       expect(mockSearchTopArtists).toHaveBeenCalledWith(
         {
             page: 1,
             country: 'spain'
         }
       );
       component.unmount();
   });

});
