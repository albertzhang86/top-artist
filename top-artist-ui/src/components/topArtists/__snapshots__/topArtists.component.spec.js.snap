// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopArtists renders correctly 1`] = `
<div
  className="search-result-wrapper"
>
  <div
    className="row"
  >
    <div
      className="col-xs-12 form-group"
    >
      <input
        className="form-control"
        defaultValue="china"
        onChange={[Function]}
        placeholder="Enter Country"
        type="text"
      />
    </div>
  </div>
  <div
    className="row search-result-container"
  >
    <div
      className="col-xs-12"
    >
      <ul
        className="list-group"
      >
        <li
          className="list-group-item row"
        >
          <div
            className="result-item row"
          >
            <div
              className="col-xs-12 col-sm-4"
            >
              <a
                className="thumbnail"
                onClick={[Function]}
                style={Object {}}
              >
                <img
                  alt="top albums of ar1"
                  src={undefined}
                />
              </a>
            </div>
            <div
              className="col-xs-12 col-sm-8"
            >
              <h3>
                ar1
              </h3>
              <p>
                <strong>
                  Listener: 
                </strong>
                2
              </p>
            </div>
          </div>
        </li>
        <li
          className="list-group-item row"
        >
          <div
            className="result-item row"
          >
            <div
              className="col-xs-12 col-sm-4"
            >
              <a
                className="thumbnail"
                onClick={[Function]}
                style={Object {}}
              >
                <img
                  alt="top albums of ar2"
                  src={undefined}
                />
              </a>
            </div>
            <div
              className="col-xs-12 col-sm-8"
            >
              <h3>
                ar2
              </h3>
              <p>
                <strong>
                  Listener: 
                </strong>
                3
              </p>
            </div>
          </div>
        </li>
      </ul>
      <div
        className="row"
      >
        <ul
          className="pagination"
        >
          <li
            className="disabled"
            style={undefined}
          >
            <a
              href="#"
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "pointerEvents": "none",
                }
              }
              tabIndex={-1}
            >
              <span
                aria-label="First"
              >
                «
              </span>
            </a>
          </li>
          <li
            className="disabled"
            style={undefined}
          >
            <a
              href="#"
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "pointerEvents": "none",
                }
              }
              tabIndex={-1}
            >
              <span
                aria-label="Previous"
              >
                ‹
              </span>
            </a>
          </li>
          <li
            className="active"
            style={undefined}
          >
            <a
              href="#"
              onClick={[Function]}
              role="button"
            >
              1
            </a>
          </li>
          <li
            className=""
            style={undefined}
          >
            <a
              href="#"
              onClick={[Function]}
              role="button"
            >
              2
            </a>
          </li>
          <li
            className=""
            style={undefined}
          >
            <a
              href="#"
              onClick={[Function]}
              role="button"
            >
              3
            </a>
          </li>
          <li
            className=""
            style={undefined}
          >
            <a
              href="#"
              onClick={[Function]}
              role="button"
            >
              <span
                aria-label="Next"
              >
                ›
              </span>
            </a>
          </li>
          <li
            className=""
            style={undefined}
          >
            <a
              href="#"
              onClick={[Function]}
              role="button"
            >
              <span
                aria-label="Last"
              >
                »
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
